mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageMapper.java

```diff
@@ -267,34 +267,37 @@ public ModSeq getHighestModSeq(Mailbox mailbox) throws MailboxException {
     public MessageMetaData add(Mailbox mailbox, MailboxMessage message) throws MailboxException {
         CassandraId mailboxId = (CassandraId) mailbox.getMailboxId();
 
-        save(mailbox, addUidAndModseq(message, mailboxId))
-            .thenEmpty(indexTableHandler.updateIndexOnAdd(message, mailboxId))
-            .block();
+        block(addUidAndModseq(message, mailboxId)
```

Comment:
should be replaced with an `onError*` method

Reply:
not that easy, I'm not aware block can thorw checked exceptions.

Reply:
Oh you are right